crates/sargon/src/system/sargon_os/sargon_os.rs

```diff
@@ -151,6 +144,51 @@ impl SargonOS {
         Ok(())
     }
 
+    pub async fn new_wallet_with_derived_bdfs(
+        &self,
+        hd_factor_source: PrivateHierarchicalDeterministicFactorSource,
+        accounts: Accounts,
+    ) -> Result<()> {
+        debug!("Deriving Profile from BDFS");
+
+        let hd_keys: Vec<HierarchicalDeterministicPublicKey> = accounts
+            .iter()
+            .map(|account| {
+                account
+                    .security_state
+                    .into_unsecured()
+                    .map(|c| c.transaction_signing.public_key)
+                    .map_err(|_| CommonError::EntitiesNotDerivedByFactorSource)
+            })
+            .try_collect()?;
```

Comment:
cool, I did not know about `try_collect`, we should embrace it!

I think I'd prefer `let ..:  Vec<T> = ....try_collect()?` be rewritten as `let .. = ....try_collect::<Vec<_>>()?`

where `try_collect::<Vec<_>>()` is verbatim, the `_` usage tells rust to fill it in (which it can do based on `.map(|c| c.transaction_signing.public_key)`.

That is the code style I've mostly used elsewhere in Sargon - the advantage is that for long chains of functional calls one can control the return type at the end of the chain, which one typically does by adding yet another map that makes it a bit easier to maintain IMO.

Reply:
I like doing this too, but this does not compile. I cannot find a sane way to make this compile.

Reply:
An example of this working is this:
```
.try_collect::<_, Vec<_>, _>()?;
```
Does this look more readable?